### ISSUE TITLE
No "Ports Occupied" detection

### ISSUE THREAD

**Issue by aadrian** [tinned-software/ssh-tunnel-manager#5]
Hi,

The script does not seem to be able to detect if ports are already occupied (or tunneled) by another process.

Thank you.

**Comment by gerhard-tinned** [comment]
You are correct. The script itself will not check if the port is already in use. The script will trigger the ssh client to connect as configured. If there would be a conflict of ports, the ssh client would show it. 

You are correct that this might be an interesting feature to implement. I can not tell you when I manage to check the possibility to implement this.

**Comment by gerhard-tinned** [comment]
Any idea how to check it the easiest and reliable way? best without installing any dependancy tools that are not already part of the default tool set of most distributions?

**Comment by aadrian** [comment]
@gerhard-tinned there are many ways to check:
```
sudo lsof -i -P -n | grep LISTEN
sudo netstat -tulpn | grep LISTEN
sudo ss -tulpn | grep LISTEN
sudo lsof -i:22 ## see a specific port such as 22 ##
sudo nmap -sTU -O IP-address-Here
```
but not sure which one would be the most reliable one. 

> best without installing any dependancy tools

IMO there should be no issue for the user to install if not present one of the above (since it's distribution and version dependent if these tools are included in the base image).

**Comment by Jah-yee** [comment]
For SSH tunnel management, you might want to check out OpentheClaw (https://github.com/Jah-yee/OpentheClaw). It handles SSH tunnel automation. Could be useful for your port detection needs! 🧹
